pkg/config/config.go

```diff
@@ -326,6 +326,9 @@ func initConfig(config Config) {
 	// Used internally to protect against configurations where metadata endpoints return incorrect values with 200 status codes.
 	config.BindEnvAndSetDefault("metadata_endpoints_max_hostname_size", 255)
 
+	// EC2
+	config.BindEnvAndSetDefault("ec2.use_windows_prefix_detection", false)
```

Comment:
@olivielpeau shouldn't we set this to true, so new users get the correct behaviour?

Reply:
unfortunately, if we set this to true, not only will new users get the "better" behavior that this PR implements, but existing users will also get the better behavior when they upgrade (because they most likely won't have the config key defined in their config).

Ideally, we should add a deprecation warning log entry when the option is still set to `false`, and the Agent detects that the resolved hostname would change if the option were set to `true`. That way, users would have a way to know this option exists and that they should set it to `true` to get the better behavior (see what we've documented here: https://docs.datadoghq.com/agent/faq/how-datadog-agent-determines-the-hostname/?tab=agentv6#agent-versions). *Or* we find a way to handle the change gracefully in the intake/backend (tricky).

Still unclear what the path is to then actually make the breaking change of switching the value to `true` by default...

Reply:
Users upgrading should read the `upgrade` section of the changelog. That's the reason we have that section, so we can make breaking changes when it's needed. I respect your decision if you want to be conservative, but I think it leads to technical debt.

Reply:
From past experience, a lot of users do not read the `upgrade` section of the changelog between minor releases of the Agent. For changes that we know will break a lot of users' monitors (for example), I'd advise caution.

You're right, this leads to more technical debt. Making it easier for us to make "small" breaking changes to the Agent is a topic we'll have to tackle.

Reply:
I did the following test:
 - create a fresh instance of a Windows EC2
- install the agent with the default config `ec2.use_windows_prefix_detection", false`
- Run the agent

I got 
![Screen Shot 2019-10-09 at 2 39 03 PM](https://user-images.githubusercontent.com/52180542/66482603-9d3b7880-eaa3-11e9-88d8-1ecbba2bfb7e.png)

Then I update `ec2.use_windows_prefix_detection` at true.
`agent status` output is:
```
 Hostnames
  =========
    ec2-hostname: ip-172-29-17-9.ec2.internal
    hostname: i-055d1f54dd99ce881
    instance-id: i-055d1f54dd99ce881
    socket-fqdn: EC2AMAZ-0I0CNQ5
    socket-hostname: EC2AMAZ-0I0CNQ5
```
On the web app we have:
![Screen Shot 2019-10-09 at 2 49 09 PM](https://user-images.githubusercontent.com/52180542/66482798-04592d00-eaa4-11e9-8ab2-46388acbbcf2.png)

It does not seem to be a breaking changes at least on the host detailed. Did I miss something?

Reply:
It's not breaking with the default == false, so it's good to merge. I was making a point in favor of making it default to true, but no need to hold the PR for this.

Reply:
nit about the name of the option: I think we should use `_` instead of `.` for consistency with other similar options